inst/tests/tests.Rraw

```diff
@@ -8563,16 +8563,24 @@ test(1600.2, names(DT1[DT2, .(id1=id1, val=val, bla=sum(z1, na.rm=TRUE)), on="id
 # warn when merge empty data.table #597
 DT0 = data.table(NULL)
 DT1 = data.table(a=1)
+
+# Test 1601.1: Merge DT1 with itself on column 'a'
 test(1601.1, merge(DT1, DT1, by="a"), data.table(a=1, key="a"))
+
+# Test 1601.2: Merge DT1 with DT0 on column 'a'
 test(1601.2, merge(DT1, DT0, by="a"),
      warning="Input data.table 'y' has no columns.",
-     error="Elements listed in `by`")
+     error="The following columns are missing:\n - From `y`: a")
+
+# Test 1601.3: Merge DT0 with DT1 on column 'a'
 test(1601.3, merge(DT0, DT1, by="a"),
      warning="Input data.table 'x' has no columns.",
-     error="Elements listed in `by`")
+     error="The following columns are missing:\n - From `x`: a")
+
+# Test 1601.4: Merge DT0 with DT0 on column 'a'
 test(1601.4, merge(DT0, DT0, by="a"),
      warning="Neither of the input data.tables to join have columns.",
-     error="Elements listed in `by`")
+     error="The following columns are missing:\n - From `x`: a\n - From `y`: a")
```

Comment:
please remove newlines in error messages

Reply:
Remove, or add? I find the current output hard to read:

```
The following columns are missing: - From x: a
```

I would find this much more readable (possibly indenting the second line):

```
The following columns are missing:
- From x: a
```

At a higher level, I wonder if translation would be easier if we instead structured the message like so:

```
The following columns are missing from x: ...
The following columns are missing from y: ...
```

Reply:
The longer sentence would definitely be easier to translate.